Yoast/javascript
# P2 506 add recommended blocks to sidebar

## Summary


This PR can be summarized in the following changelog entry:

* Adds a panel to the sidebar for Recommended blocks.

## Relevant technical choices:

* this block shares most of its behavior with Required blocks, by design. Both deliver a panel to the sidebar which allows blocks to be added.
* The Innerblocks.ts file had to be renamed to Innerblocks.tsx, as we had to manipulate the React Elements in the Sidebar function.
* extended unit tests for the process function to guarantee proper parsing of recommended vs required blocks.
* add any missing entries from required and recommended blocks to the allowed blocks list.

## Test instructions

This PR can be tested by following these steps:

change recipe.block.php to the following snippet:
```<?php // phpcs:ignore Internal.NoCodeFound ?>
{{block name="yoast/recipe" title="Recipe" category="common" }}
{{sidebar-duration name="cook-time" output=false label="Cook time" }}
{{sidebar-duration name="prep-time" output=false label="Preparation time" }}
{{sidebar-input name="yield" output=false type="number" label="Serves #" }}
<div class={{class-name}}>
	{{variable-tag-rich-text name="title" tags=[ "h1", "h2", "h3", "h4", "h5", "h6", "strong" ] placeholder="Recipe title" }}
	{{inner-blocks allowed-blocks=[ "core/image", "core/paragraph" ] required-blocks=[ { "name": "yoast/steps", "option": "One" } , { "name": "yoast/ingredients", "option": "One" } ] recommended-blocks=[ {"name": "core/image" }, { "name": "core/paragraph" }] appender="button" appenderLabel="Add to recipe" }}
</div>

```
* in the monorepo/src/schema-blocks folder: `yarn install` and `yarn build`
* in the plugin: `grunt build:js`
* add a recipe to a post. 
* notice the sidebar contains required blocks and recommended blocks.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * code is hidden behind the schema blocks feature. 

## UI changes

## Quality assurance


Fixes #
